src/models/state/mod.rs

```diff
@@ -725,88 +730,75 @@ impl GlobalState {
-        triton_vm_job_queue: &TritonVmJobQueue,
-    ) -> Result<(Transaction, Option<TxOutput>)> {
-        self.create_transaction_with_prover_capability(
-            tx_outputs,
-            change_key,
-            change_utxo_notify_medium,
-            fee,
-            timestamp,
-            self.proving_capability(),
-            triton_vm_job_queue,
-        )
-        .await
-        .map(|(tx, _tx_details, output)| (tx, output))
-    }
-
-    /// Variant of [Self::create_transaction] that allows caller to specify
-    /// prover capability. [Self::create_transaction] is the preferred interface
-    /// for anything but tests.
-    #[expect(clippy::too_many_arguments)]
-    pub(crate) async fn create_transaction_with_prover_capability(
+    pub(crate) async fn create_transaction(
         &self,
         mut tx_outputs: TxOutputList,
-        change_key: SpendingKey,
-        change_utxo_notify_medium: UtxoNotificationMedium,
         fee: NativeCurrencyAmount,
         timestamp: Timestamp,
-        prover_capability: TxProvingCapability,
-        triton_vm_job_queue: &TritonVmJobQueue,
-    ) -> Result<(Transaction, TransactionDetails, Option<TxOutput>)> {
-        // TODO: Attempt to simplify method interface somehow, maybe by moving
-        // it to GlobalStateLock?
+        config: TxCreationConfig,
+    ) -> Result<TxCreationArtifacts> {
         let tip = self.chain.light_state();
         let tip_mutator_set_accumulator = tip.mutator_set_accumulator_after();
         let tip_digest = tip.hash();
 
-        // 1. create/add change output if necessary.
-        let total_spend = tx_outputs.total_native_coins() + fee;
+        // 1. balance transaction
+        let total_cost = tx_outputs.total_native_coins() + fee;
 
         // collect spendable inputs
         let tx_inputs = self
             .wallet_state
             .allocate_sufficient_input_funds(
-                total_spend,
+                total_cost,
                 tip_digest,
                 &tip_mutator_set_accumulator,
                 timestamp,
             )
             .await?;
 
-        let total_spendable = tx_inputs
+        let total_unlocked = tx_inputs
             .iter()
             .map(|x| x.utxo.get_native_currency_amount())
             .sum();
 
-        // Add change, if required to balance tx.
-        let mut maybe_change_output = None;
-        if total_spend < total_spendable {
-            let amount = total_spendable.checked_sub(&total_spend).ok_or_else(|| {
-                anyhow::anyhow!("overflow subtracting total_spend from input_amount")
-            })?;
-
-            let change_utxo =
-                self.create_change_output(amount, change_key, change_utxo_notify_medium)?;
-            tx_outputs.push(change_utxo.clone());
-            maybe_change_output = Some(change_utxo);
-        }
+        // Add change output, if required to balance transaction
+        let nonzero_change = total_cost < total_unlocked;
```

Comment:
`nonzero_change` sounds like it's of type `NonZero<Change>` or similar.

```suggestion
        let change_is_nonzero = total_cost < total_unlocked; // or “is_nonzero_change”
```